src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -2038,6 +2053,9 @@ private void Window_PreviewMouseLeftButtonUp(object sender, MouseButtonEventArgs
 
         public void Dispose()
         {
+            //TODO we should be careful with this, viewExtensions also have their shutdown method called
+            //when the dynamoWindow closes so it's possible there are some double dispose issues for ours or
+            //external extensions. Looking for feedback here!
```

Comment:
For the extensions I have seen we usually call `Dispose` from `Shutdown`. I'm not sure there is any documentation established around that pattern though.

You might want to check in which cases, `DynamoView.Dispose` was being called directly, and maybe use another `DisposeAllButExtensions` in the context of `WindowClosed`. In any case this is not so bad as any error will be catched.

Reply:
DynamoView.Dispose was never called!

Reply:
Could integrators be using it? If not, maybe we can remove the part where we `Dispose` extensions, just to keep things the way they were before.

Reply:
that was a possible solution, the safest thing to do for now is to leave out the ViewExtension dispose calls and properly document the relationship between shutdown and dispose.